src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -330,12 +330,13 @@ trait GoldTransforms extends SparkSessionWrapper {
           $"clusterPotential.unixTimeMS_state_start"
             .between($"workerNodeDetails.activeFromEpochMillis", $"workerNodeDetails.activeUntilEpochMillis"),
         "left")
-      .drop("activeFrom", "activeUntil", "activeFromEpochMillis", "activeUntilEpochMillis", "activeFromDate", "activeUntilDate", "worker_orgid", "node_type_id_lookup")
+      .withColumn("current_num_workers",when((col("current_num_workers") === "0" && col("num_workers").isNotNull) ,col("num_workers")).otherwise(col("current_num_workers")))
+      .withColumn("target_num_workers",when((col("target_num_workers") === "0" && col("num_workers").isNotNull),col("num_workers")).otherwise(col("target_num_workers")))
```

Comment:
Are you sure that at this point in the DF current_num_workers will be 0 or will it be null? Also, this is going to adjust ALL records that have current/target_num_workers from 0 to whatever is defined in the snapshot. If null, then we can probably just coalesce in all cases but if it truly is 0 then additional logic is likely necessary to make sure we don't convert true 0s to the snapshot config value.

Overall, I'm concerned that this is not the right place to apply this logic. I believe we should apply the appropriate logic in clusterSpec so that this logic ONLY HAPPENS on FIRST RUN. Doing this every run will likely lead to unexpected overwrites (e.g. single nodes).